### PR TITLE
Resource manager fixes

### PR DESCRIPTION
**Pull request by hkaiser**
A couple of fixes I had lying around for a while in my local copy of the repository.

This improves the behavior of the resource-manager in cases where the user requests more resources than available.

### CODE REVIEW

File: tests/unit/threads/resource_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        HPX_TEST_EQ(
+            infotraits::processing_units_count(shrink_exec, test::dummy),
+            processing_units - num_execs * num_pus);
+
+        // schedule a couple of tasks on each of the executors
+        for(Executor& exec : execs)
+        {
+            for (int i = 0; i != num_tasks; ++i)
+            {
+                traits::apply_execute(exec,
```
Comment (sithhell) [inline tests/unit/threads/resource_manager.cpp]: could this use the new executor interface instead?

File: tests/unit/threads/resource_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            HPX_TEST_EQ(infotraits::processing_units_count(exec, test::dummy),
+                num_pus);
+        }
+
+        // schedule a couple of tasks on each of the executors
+        for(Executor& exec : execs)
+        {
+            for (int i = 0; i != num_tasks; ++i)
+            {
+                traits::apply_execute(exec,
```
Comment (sithhell) [inline tests/unit/threads/resource_manager.cpp]: could this use the new executor interface instead?

Comment (hkaiser) [inline tests/unit/threads/resource_manager.cpp]: This code will go away once this PR is merged to master, it's old code not touched by this PR.

File: tests/unit/threads/resource_manager.cpp (hunk 1 above)
Comment (hkaiser) [inline tests/unit/threads/resource_manager.cpp]: Same as above.

**Comment by hkaiser** [comment]
@sithhell Your comments have been addressed.

**Pull request merged**